lib/puppet/application/ssl.rb

```diff
@@ -130,4 +138,16 @@ def verify(host)
     puts "Verify failed: #{e.message}"
     exit(1)
   end
+
+  def clean(host)
+    settings = %w[hostprivkey hostpubkey hostcsr hostcert passfile]
+    settings.concat(%w[localcacert hostcrl]) if options[:local_ca]
+    settings.each do |setting|
+      path = Puppet[setting]
+      if Puppet::FileSystem.exist?(path)
+        Puppet::FileSystem.unlink(path)
+        puts "Deleted #{path}"
```

Comment:
Why again can't we use Puppet.info here?

Reply:
The short answer is because `puppet ssl` uses `puts` elsewhere.

The long answer is... We could use `Puppet.{notice,info,etc}`. The good thing is colorization, and it would match `puppet cert clean` display:

```
$ bx puppet cert clean foobar
Warning: `puppet cert` is deprecated and will be removed in a future release.
   (location: /Users/josh/work/puppet/lib/puppet/application.rb:370:in `run')
Notice: Revoked certificate with serial 3
Notice: Removing file Puppet::SSL::Certificate foobar at '/Users/josh/.puppetlabs/etc/puppet/ssl/ca/signed/foobar.pem'
Notice: Removing file Puppet::SSL::Certificate foobar at '/Users/josh/.puppetlabs/etc/puppet/ssl/certs/foobar.pem'
Notice: Removing file Puppet::SSL::Key foobar at '/Users/josh/.puppetlabs/etc/puppet/ssl/private_keys/foobar.pem'
```
The bad thing IMO is it prepends the message type and the message may be excluded based on the current log level, though that could be a good thing for `Puppet.debug`...

I'm happy to switch things over to using puppet's logger, though I think we should file a separate ticket for that.

Reply:
Fair, I thought there was a point where we couldn't because the logger wasn't set up yet or something like that. I was just curious, I'm happy to defer to you regarding logging best practices (:

Reply:
ah, yeah you have to force the `:console` log destination, but that's easy for CLIs that always write to stdout/stderr